crosschain-resolver/contracts/L1Resolver.sol

```diff
@@ -218,12 +256,42 @@ contract L1Resolver is EVMFetchTarget, ITargetResolver, IExtendedResolver, ERC16
         return abi.encode(values[1]);
     }
 
+    /**
+     * @notice Get metadata about the L1 Resolver
+     * @dev This function provides metadata about the L1 Resolver, including its name, coin type, GraphQL URL, storage type, and encoded information.
+     * @param name The domain name in format (dnsEncoded)
+     * @return name The name of the resolver ("CCIP RESOLVER")
+     * @return coinType Resolvers coin type (60 for Ethereum)
+     * @return graphqlUrl The GraphQL URL used by the resolver
+     * @return storageType Storage Type (0 for EVM)
+     * @return storageLocation The storage identifier. For EVM chains, this is the address of the resolver contract.
+     * @return context can be l2 resolver contract address for evm chain but can be any l2 storage identifier for non-evm chain
+     */
+    function metadata(
+        bytes calldata name
+    ) public view returns (string memory, uint256, string memory, uint8, bytes memory, bytes memory) {
+        (, address target) = getTarget(name);
+
+        return (
+            resolverName,
+            l2ResolverCoinType,
+            graphqlUrl,
+            uint8(0), // storage Type 0 => EVM
+            abi.encodePacked(address(target)), // storage location => l2 resolver address
+            abi.encodePacked(address(target))  // context => l2 resolver address
+        );
+    }
+    function id() public pure returns(bytes4){
```

Comment:
What is this for?

Reply:
I forgot to remove it after obtaining the functionid. I will remove it